src/cascade/model/var.py

```diff
@@ -0,0 +1,109 @@
+import numpy as np
+from scipy.interpolate import RectBivariateSpline, interp1d
+
+from cascade.model.age_time_grid import AgeTimeGrid
+
+
+class Var(AgeTimeGrid):
+    """A Var is a set of values of a random field on a SmoothGrid.
+
+    It's an AgeTimeGrid, but AgeTimeGrids present each data item
+    as a row. This presents each data item as a float. It also
+    behaves like a continuous function of age and time.
+    """
+    def __init__(self, age_time_grid):
+        super().__init__(age_time_grid, columns=["mean"])
+        self._spline = None
+
+    def check(self, name=None):
+        """None of the means should be nan. There should only be the
+        three mulstds."""
+        if not self.grid["mean"].notna().all():
+            raise RuntimeError(
+                f"Var {name} has {self.grid['mean'].isna().sum()} nan values")
+        if set(self.mulstd.keys()) - {"value", "dage", "dtime"}:
+            raise RuntimeError(
+                f"Var {name} has mulstds besides the three: {list(self.mulstd.keys())}"
+            )
+
+    def __setitem__(self, at_slice, value):
+        """
+        Args:
+            at_slice (slice, slice): What to change, as integer offset into
+                ages and times.
+            value (float): Set with a single floating-point value.
+        """
+        super().__setitem__(at_slice, [value])
+
+    def __getitem__(self, item):
+        return float(super().__getitem__(item)["mean"])
+
+    def set_mulstd(self, kind, value):
+        self.mulstd[kind].loc[:, "mean"] = value
+
+    def get_mulstd(self, kind):
+        return float(self.mulstd[kind]["mean"])
+
+    def __str__(self):
+        return f"Var({len(self.ages), len(self.times)})"
+
+    def __call__(self, age, time):
+        """Call a Var as a function of age and time.
```

Comment:
I really like this interface.

Reply:
OMG, yes. It makes something that seemed hard, "bivariate interpolation given a Pandas DataFrame", and turn it into an automatic tool. Covers over complexity in ensuring that grids match each other, too.